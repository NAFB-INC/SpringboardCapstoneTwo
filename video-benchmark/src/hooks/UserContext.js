import React from "react";

//storage place for app-wide data
const UserContext = React.createContext();

export default UserContext;
