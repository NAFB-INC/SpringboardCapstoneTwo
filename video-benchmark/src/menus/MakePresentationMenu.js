import React from "react";

function MakePresentationMenu() {

    return (
        <div className="MakePresentationMenu">
            MakePresentationMenu
        </div>
    );
}

export default MakePresentationMenu;