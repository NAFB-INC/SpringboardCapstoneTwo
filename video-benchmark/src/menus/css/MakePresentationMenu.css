.MakePresentationMenu{
    position:absolute;
    margin-top:5%;
    margin-left:5%;
    margin-right:5%;
    padding-top:2%;
    height:105%;
    width:88%;
    border:5px solid #6D2E46;
    background-color: #ECE2D0;
}