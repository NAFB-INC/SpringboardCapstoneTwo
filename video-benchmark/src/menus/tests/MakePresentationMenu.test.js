import React from "react";
import { render } from '@testing-library/react';
import MakePresentationMenu from '../MakePresentationMenu';
import UserContext from "../../hooks/UserContext";

let baseURL = "hi";
const testCase1 = (
  <div>
      <UserContext.Provider value={{ baseURL }}>
          <MakePresentationMenu />
      </UserContext.Provider>
  </div>);


it("renders without crashing", function() {
  render(testCase1);
});

it("matches snapshot", function() {
  const { asFragment } = render(testCase1);
  expect(asFragment()).toMatchSnapshot();
});