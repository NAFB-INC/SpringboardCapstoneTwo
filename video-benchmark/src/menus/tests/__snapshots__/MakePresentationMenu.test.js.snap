// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="MakePresentationMenu"
    >
      <div
        class="PresentationForm"
      >
        <div
          class="form-header"
        >
          Let's create a new presentation!
        </div>
        <div
          class="info-box"
        >
          <p>
            Once created, you can access your presentation using the presentation code you will be given, and the password you set. Don't lose it!
          </p>
          <p>
            You can start your presentation at any time in the next 
            <b>
              30 days
            </b>
             without needing to create a new one.
          </p>
          <p>
            All you need to do is send that presentation code (or a direct audience link) to anyone you want to join!
          </p>
        </div>
        <form>
          <div
            class="form-box"
          >
            <label
              for="presentation_title"
            >
              Presentation Title: 
            </label>
            <input
              autocomplete="presentation-title"
              name="presentation_title"
              placeholder="Major Budget Overhaul"
              type="text"
              value=""
            />
          </div>
          <div
            class="form-box"
          >
            <label
              for="presenter_name_first"
            >
              Presenter's First Name: 
            </label>
            <input
              autocomplete="first-name"
              name="presenter_name_first"
              placeholder="Joe"
              type="text"
              value=""
            />
          </div>
          <div
            class="form-box"
          >
            <label
              for="presenter_name_last"
            >
              Presenter's Last Name: 
            </label>
            <input
              autocomplete="last-name"
              name="presenter_name_last"
              placeholder="Smith"
              type="text"
              value=""
            />
          </div>
          <div
            class="form-box"
          >
            <label
              for="presenter_pass"
            >
              Presenter Password: 
            </label>
            <input
              autocomplete="presenter-password"
              name="presenter_pass"
              placeholder="super secret"
              type="password"
              value=""
            />
          </div>
          <div>
            <small>
              This will ensure only you can start the presentation. For demo purposes, do NOT use a secure password. Password security planned in future update.
            </small>
          </div>
          <div
            class="form-box"
          >
            <label
              for="presentation_description"
            >
              [OPTIONAL] Presentation Description: 
            </label>
            <textarea
              autocomplete="presentation-description"
              name="presentation_description"
              placeholder="Something needs to be done about our 1st class travel spending..."
              type="text"
            />
          </div>
          <br />
          <div
            class="errors"
          />
          <button
            class="submit-button"
          >
            <div
              class="container"
            >
              <div
                class="tick"
              >
                Submit
              </div>
            </div>
          </button>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
